### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by dgageot**

### CODE REVIEW

File: pkg/skaffold/util/util.go [hunk 1]
```diff
@@ -189,9 +168,9 @@ func VerifyOrCreateFile(path string) error {
 
 // RemoveFromSlice removes a string from a slice of strings
 func RemoveFromSlice(s []string, target string) []string {
-	for i, val := range s {
-		if val == target {
-			return append(s[:i], s[i+1:]...)
+	for i := len(s) - 1; i >= 0; i-- {
+		if s[i] == target {
+			s = append(s[:i], s[i+1:]...)
```
Comment (nkubala) [inline pkg/skaffold/util/util.go]: :flushed: was this causing bugs anywhere?

File: pkg/skaffold/runner/dev.go [hunk 2]
```diff
@@ -91,8 +91,7 @@ func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*la
 	// Watch artifacts
 	for i := range artifacts {
 		artifact := artifacts[i]
-
-		if !r.IsTargetImage(artifact) {
+		if !r.runCtx.Opts.IsTargetImage(artifact) {
```
Comment (nkubala) [inline pkg/skaffold/runner/dev.go]: this was broken before then?

Comment (dgageot) [reply pkg/skaffold/runner/dev.go]: No. I just moved the function

File: pkg/skaffold/util/util.go (hunk 1 above)
Comment (dgageot) [reply pkg/skaffold/util/util.go]: I don’t think so. I added tests and realize it was doing the job only partially

**Comment by priyawadhwa** [comment]
This needs a rebase :)

**Comment by dgageot** [comment]
@nkubala @priyawadhwa It has been rebased

**Pull request merged**